projecttacoma/fqm-execution
# Track valueComparisonLocalId for isNull filter

# Summary
Updated the interpretExpression code to track valueComparisonLocalId for isNull filter

## New behavior
The isNull expression could be conducting further filtering on the result of a query. To identify gaps in care, we want to track the id of the isNull expression down the expression stack so we can reference it if the underlying query winds up causing the gap. This PR adds valueComparisonLocalId tracking for isNull filters, and updates the dataRequirements to track is null filters as well.

## Code changes
- Added valueComparisonLocalId tracking to isNull filter when parsing isNull expression
- Updated the generateDetailedValueFilters function to handle isNull filters when creating dataRequirements filters
- Fixed typos ;)
- Added unit testing

# Testing guidance
- Run `npm test`
- Try running a queryInfo calculation on the HbA1c measure (ping me if you don't have it), and ensure that the is null query filter in the resulting JSON tracks the valueComparisonLocalId
- Try running dataRequirements on the HbA1c measure and ensure that there now exists a dataRequirement with value filter "null" on an Observation resource from the "http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.198.12.1013" valueset
